packages/cardano-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/DbSyncStakePoolSearch.ts

```diff
@@ -0,0 +1,255 @@
+/* eslint-disable sonarjs/no-nested-template-literals */
+import { Cardano, MultipleChoiceSearchFilter, StakePoolQueryOptions, StakePoolSearchProvider } from '@cardano-sdk/core';
+import {
+  EpochModel,
+  // EpochRewardModel,
+  OwnerAddressModel,
+  PoolDataModel,
+  PoolRegistrationModel,
+  PoolRetirementModel,
+  PoolUpdateModel,
+  RelayModel,
+  SubQuery
+} from './types';
+import { Logger, dummyLogger } from 'ts-log';
+import { Pool, QueryResult } from 'pg';
+import {
+  mapAddressOwner,
+  // mapEpochReward,
+  mapPoolData,
+  mapPoolRegistration,
+  mapPoolRetirement,
+  mapPoolUpdate,
+  mapRelay,
+  toCoreStakePool
+} from './mappers';
+import Queries, {
+  addSentenceToQuery,
+  buildOrQueryFromClauses,
+  findLastEpoch,
+  getIdentifierFullJoinClause,
+  getIdentifierWhereClause,
+  getStatusWhereClause,
+  poolsByPledgeMetSubqueries,
+  withPagination
+} from './queries';
+
+export class DbSyncStakePoolSearchProvider implements StakePoolSearchProvider {
+  #db: Pool;
+  #logger: Logger;
+
+  constructor(db: Pool, logger = dummyLogger) {
+    this.#logger = logger;
+    this.#db = db;
+  }
+
+  private async queryRetirements(hashesIds: number[]) {
+    this.#logger.debug('About to query pool retirements');
+    const result: QueryResult<PoolRetirementModel> = await this.#db.query(Queries.findPoolsRetirements, [hashesIds]);
+    return result.rows.length > 0 ? result.rows.map(mapPoolRetirement) : [];
+  }
+  private async queryRegistrations(hashesIds: number[]) {
+    this.#logger.debug('About to query pool registrations');
+    const result: QueryResult<PoolRegistrationModel> = await this.#db.query(Queries.findPoolsRegistrations, [
+      hashesIds
+    ]);
+    return result.rows.length > 0 ? result.rows.map(mapPoolRegistration) : [];
+  }
+  private async queryPoolRelays(updatesIds: number[]) {
+    this.#logger.debug('About to query pool relays');
+    const result: QueryResult<RelayModel> = await this.#db.query(Queries.findPoolsRelays, [updatesIds]);
+    return result.rows.length > 0 ? result.rows.map(mapRelay) : [];
+  }
+  private async queryPoolOwners(hashesIds: number[]) {
+    this.#logger.debug('About to query pool owners');
+    const result: QueryResult<OwnerAddressModel> = await this.#db.query(Queries.findPoolsOwners, [hashesIds]);
+    return result.rows.length > 0 ? result.rows.map(mapAddressOwner) : [];
+  }
+  // private async queryPoolRewards(hashesIds: number[], limit?: number) {
```

Comment:
Just realised this is still incomplete. Are you blocked here @jvieiro ?

Reply:
Ive created a separate branch for this. [Here](https://github.com/input-output-hk/cardano-js-sdk/blob/feat/add-rewards-on-stake-pool-http-server/packages/cardano-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/queries.ts#L13) is the query. Im still missing [epoch length in milliseconds](https://github.com/input-output-hk/cardano-js-sdk/blob/feat/add-rewards-on-stake-pool-http-server/packages/core/src/Cardano/types/StakePool/StakePool.ts#L56) which Im not pretty sure on how to calculate it, but I think I will use the difference between the [time](https://github.com/input-output-hk/cardano-db-sync/blob/master/doc/schema.md#block) of the first and last block of each epoch. Then, I have to add the data-mappers and adapt the tests with the rewards

Reply:
I think it comes from shelley genesis as `"slotLength": 1` (1 second per slot) + `"epochLength": 432000` (that many slots)

Reply:
That would work for shelley but for other eras would be another values, right? Ive recently figured out that `epoch` table has `start_time` and `end_time`, so with this it should be enough. Ref https://github.com/input-output-hk/cardano-db-sync/blob/master/doc/schema.md#epoch

Reply:
It hasn't changed since shelley and is not part of protocol parameters update, so it should be safe to use static values from genesis. Selecting it from db-sync works too, just slightly less efficient I guess